File: atelier_front/lib/infrastructure/api/creative.ts
import type { Asset, AssetKind } from '@/lib/domain/asset'
import { api } from '@/lib/infrastructure/http/client'

export const creativeApi = {
  generate(input: { kind: AssetKind; prompt: string }): Promise<Asset> {
    return api<Asset>('/api/v1/creative/generate', {
      method: 'POST',
      body: JSON.stringify(input),
    })
  },
  list(): Promise<Asset[]> {
    return api<Asset[]>('/api/v1/creative/assets')
  },
}

File: atelier_backend/app/api/v1/creative.py
from fastapi import APIRouter

from app.api.deps import CreativeServiceDep, CreatorUser
from app.api.schemas import AssetResponse, GenerateRequest

router = APIRouter(prefix="/creative", tags=["creative"])


@router.post("/generate")
async def generate_asset(
    body: GenerateRequest,
    user: CreatorUser,
    service: CreativeServiceDep,
) -> AssetResponse:
    asset = await service.generate(user, kind=body.kind, prompt=body.prompt)
    return AssetResponse.from_entity(asset)


@router.get("/assets")
async def list_assets(user: CreatorUser, service: CreativeServiceDep) -> list[AssetResponse]:
    _ = user
    assets = await service.list_assets()
    return [AssetResponse.from_entity(asset) for asset in assets]
